src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -53,63 +55,142 @@ public static IEnumerable<DockerContainerInstance> GetLocalDockerContainers(stri
                 {
                     if (!string.IsNullOrWhiteSpace(args))
                     {
-                        if (args.Trim()[0] != '{')
-                        {
-                            // output isn't json, command Error
-                            errorSB.Append(args);
-                        }
-                        else
-                        {
-                            if (DockerContainerInstance.TryCreate(args, out DockerContainerInstance containerInstance))
-                            {
-                                containers.Add(containerInstance);
-                            }
-                            containerCount++;
-                        }
+                        args = args.Trim(charsToTrim);
+                        callback(args);
                     }
                 });
 
                 commandRunner.Start();
 
                 bool cancellationRequested = false;
-                VS.VSOperationWaiter.Wait(UIResources.QueryingForContainersMessage, false, (cancellationToken) =>
+                VS.VSOperationWaiter.Wait(UIResources.RunningDockerCommandMessage, false, (cancellationToken) =>
                 {
-                    while (!resetEvent.WaitOne(2000) && !cancellationToken.IsCancellationRequested)
-                    { }
+                    while (!resetEvent.WaitOne(2000) && !cancellationToken.IsCancellationRequested) { }
                     cancellationRequested = cancellationToken.IsCancellationRequested;
                 });
 
                 if (!cancellationRequested)
                 {
-                    // might need to throw an exception here too??
                     if (exitCode.GetValueOrDefault(-1) != 0)
                     {
-                        // if the exit code is not zero, then the output we received possibly is the error message
                         string exceptionMessage = UIResources.CommandExecutionErrorWithExitCodeFormat.FormatCurrentCultureWithArgs(
```

Comment:
So I see this message being created, but since you removed the `throw` this isn't being used anywhere. There needs to be a way to bubble up the error message.